firefox-ios/Client/Frontend/Settings/Main/Support/SendAnonymousUsageDataSetting.swift

```diff
@@ -54,6 +55,11 @@ class SendAnonymousUsageDataSetting: BoolSetting {
         self.settingDidChange = { [weak self] value in
             // AdjustHelper.setEnabled($0)
             DefaultGleanWrapper.shared.setUpload(isEnabled: value)
+
+            if !value {
+                self?.prefs?.removeObjectForKey(PrefsKeys.Usage.profileId)
```

Comment:
This should also reset the profile id, just to ensure we _really_ forget the previous one.

```suggestion
                self?.prefs?.removeObjectForKey(PrefsKeys.Usage.profileId)
                GleanMetrics.Usage.profileId.set(UUID(uuidString: "beefbeef-beef-beef-beef-beeefbeefbee"))
```